File: Python/microsoft_bonsai_api/client/bonsaiclient.py
from microsoft_bonsai_api.simulator import SimulatorAPI
from .config import BonsaiClientConfig

# The API object that handles the REST connection to the bonsai platform.
class BonsaiClient(SimulatorAPI):
    def __init__(self, config: BonsaiClientConfig, **kwargs):
        self._headers = {
            "Content-Type": "application/json",
            "Authorization": config.access_key,
        }

        self.workspace = config.workspace
        super(BonsaiClient, self).__init__(
            base_url=config.server,
            headers=self._headers,
            logging_enable=config.enable_logging,
            kwargs=kwargs,
        )

File: Typescript/lib/src/simulator/client/bonsaiClient.ts
/*
 * bonsaiClient.ts
 *
 * Client object that handles connection to the bonsai platform
 *
 * Copyright: (c) Microsoft. All rights reserved.
 */

import * as Models from '../generated/models';
import { SimulatorAPI } from '../generated/simulatorAPI';
import { BonsaiClientConfig } from './bonsaiClientConfig';
import { OperationArguments, ServiceCallback, OperationSpec, RestResponse } from '@azure/ms-rest-js';

export class BonsaiClient extends SimulatorAPI {
    _headers: { [key: string]: any };
    constructor(config: BonsaiClientConfig, options?: Models.SimulatorAPIOptions) {
        super(options);
        this._headers = {
            Authorization: config.accessKey,
        };
    }

    sendOperationRequest(
        operationArguments: OperationArguments,
        operationSpec: OperationSpec,
        callback?: ServiceCallback<any>
    ): Promise<RestResponse> {
        if (operationArguments.options === undefined) {
            operationArguments.options = {};
        }
        operationArguments.options.customHeaders = this._headers;
        return super.sendOperationRequest(operationArguments, operationSpec, callback);
    }
}
